### PR TITLE
[BEAM-14129] Clean up PubsubLiteIO by removing options that no longer apply

### PR DESCRIPTION
**Pull request by dpcollins-google**
Also remove explicit blocking on data delivery- yield to the runtime instead if no data is available since reading from Pub/Sub Lite is now performed in the background

### CODE REVIEW

**Comment by dpcollins-google** [comment]
R: @lukecwik Hi luke, could you advise on whether the change here to yield to the runtime instead of blocking for a fixed amount of time is one that increases conformance with dataflow's expectations of how DoFns operate, and is unlikely to cause other subtle issues?

**Comment by aaltay** [comment]
What is the next step on this PR?

**Comment by dpcollins-google** [comment]
@lukecwik or @aaltay could you or someone else with SDF knowledge check the statement above

**Comment by aaltay** [comment]
@lukecwik would be the right person to comment.

(I accidentally closed, and re-opened.)

**Comment by aaltay** [comment]
What is the next step on this PR?

**Comment by dpcollins-google** [comment]
Id hope @lukecwik could find time to review this at some point?

**Comment by aaltay** [comment]
> Id hope @lukecwik could find time to review this at some point?

Feel free to ping people in other channels if reviews are delayed. It is easy to miss GH notifications.

**Review by lukecwik — COMMENTED** [review]
How does this impact performance?

**Review by lukecwik — CHANGES_REQUESTED** [review]
Please fix spotless and checkstyle warnings.

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsublite/internal/SubscriptionPartitionProcessorImplTest.java [hunk 1]
```diff
@@ -170,9 +151,6 @@ public void failedClaimCausesStop() {
   @Test
   public void successfulClaimThenTimeout() {
```
[the remark this answers is not in the document]
Comment (dpcollins-google) [reply sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsublite/internal/SubscriptionPartitionProcessorImplTest.java]: Done.

**Comment by dpcollins-google** [comment]
> How does this impact performance?

I ran a load test, it has the same bottleneck as the old code in dataflow (capping out at 2GiB/s on 100 workers) but is not a performance regression.

File: sdks/python/apache_beam/io/gcp/pubsublite/external.py [hunk 2]
```diff
@@ -30,8 +30,7 @@
 
 _ReadSchema = typing.NamedTuple(
     '_ReadSchema',
```
Comment (lukecwik) [inline sdks/python/apache_beam/io/gcp/pubsublite/external.py:32]:
```suggestion
    '_ReadSchema', [('subscription_path', str), ('deduplicate', bool)])
```

File: sdks/python/apache_beam/io/gcp/pubsublite/external.py [hunk 3]
```diff
@@ -30,8 +30,7 @@
...
-    [('subscription_path', str), ('min_bundle_timeout', int),
-     ('deduplicate', bool)])
+    [('subscription_path', str), ('deduplicate', bool)])
```
Comment (lukecwik) [inline sdks/python/apache_beam/io/gcp/pubsublite/external.py:33]:
```suggestion
```

File: sdks/python/apache_beam/io/gcp/pubsublite/external.py [hunk 4]
```diff
@@ -79,7 +71,6 @@ def __init__(
         NamedTupleBasedPayloadBuilder(
             _ReadSchema(
                 subscription_path=subscription_path,
-                min_bundle_timeout=min_bundle_timeout,
                 deduplicate=deduplicate)),
```
Comment (lukecwik) [inline sdks/python/apache_beam/io/gcp/pubsublite/external.py:74]:
```suggestion
```

**Review by lukecwik — COMMENTED** [review]
Fix python formatting issues

**Review by lukecwik — APPROVED** [review]

**Comment by lukecwik** [comment]
Run Python PreCommit

**Comment by lukecwik** [comment]
Run Java PreCommit

**Comment by lukecwik** [comment]
Run Java PreCommit

**Comment by lukecwik** [comment]
Run Java PreCommit

**Comment by lukecwik** [comment]
Run Java PreCommit

**Comment by ns-shua** [comment]
@lukecwik  Have you guys done performance test? Without flow setting it is running super slow and I keep getting NO_CLIENT_TOKEN on server side, besides why do you use publisher timestamp as event timestamp, why can't that be provided by user?

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/internal/SubscriptionPartitionProcessorImpl.java [hunk 5]
```diff
@@ -59,42 +52,24 @@ class SubscriptionPartitionProcessorImpl implements SubscriptionPartitionProcess
...
+    for (Optional<SequencedMessage> next = subscriber.peek();
+        next.isPresent();
+        next = subscriber.peek()) {
+      SequencedMessage message = next.get();
+      Offset messageOffset = Offset.of(message.getCursor().getOffset());
+      if (tracker.tryClaim(OffsetByteProgress.of(messageOffset, message.getSizeBytes()))) {
+        subscriber.pop();
+        lastClaimedOffset = Optional.of(messageOffset);
+        receiver.outputWithTimestamp(
+            message, new Instant(Timestamps.toMillis(message.getPublishTime())));
```
Comment (ns-shua) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/internal/SubscriptionPartitionProcessorImpl.java:66]: Why can't user define the timestamp fn?

**Pull request merged**